include/boost/math/interpolators/cubic_b_spline.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright Nick Thompson, 2017
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// This implements the compactly supported cubic b spline algorithm described in
+// "Numerical Analsis, Graduate Texts in Mathematics 181", by Rainer Kress, section 8.3.
```

Comment:
I'm going to be very pedantic and say this reference should be complete, i.e., include the year, publisher, edition, etc and ideally be in one of the usual standard citation formats.

I think these interpolation functions are a great addition to Boost.Math, btw, thanks!